src/common/getters.ts

```diff
@@ -775,6 +442,398 @@ export function getInherit(vc: DatasetWithId): boolean {
+ * Throws on type mismatch, and returns an empty array if no values are found.
+ *
+ * @example
+ * ```
+ * const accessRequest = await issueAccessRequest({...}, {
+ *  ...,
+ *  customFields: new Set([
+ *     {
+ *       key: new URL("https://example.org/ns/customInteger"),
+ *       value: [1, 2, 3],
+ *     },
+ *   ]),
+ * });
+ * // i is [1, 2, 3]
+ * const i = getCustomIntegers(accessRequest, new URL("https://example.org/ns/customInteger"));
+ * ```
+ *
+ * @param accessCredential The Access Credential (Access Grant or Access Request)
+ * @returns the custom integer array field with the provided name
+ * @since unreleased
+ */
+export function getCustomIntegers(
+  accessCredential: DatasetWithId,
+  field: URL,
+): number[] {
+  return deserializeFields(
+    accessCredential,
+    field,
+    (i: unknown): i is number => typeof i === "number" && Number.isInteger(i),
+    "integer",
+  );
+}
+
+/**
+ * Reads the custom integer value with the provided name in the consent section of the provided Access Credential.
+ * Throws on type mismatch, and returns `undefined` if no values are found.
+ *
+ * @example
+ * ```
+ * const accessRequest = await issueAccessRequest({...}, {
+ *  ...,
+ *  customFields: new Set([
+ *     {
+ *       key: new URL("https://example.org/ns/customInteger"),
+ *       value: 1,
+ *     },
+ *   ]),
+ * });
+ * // i is 1
+ * const i = getCustomString(accessRequest, new URL("https://example.org/ns/customInteger"));
+ * ```
+ *
+ * @param accessCredential The Access Credential (Access Grant or Access Request)
+ * @returns the custom integer field with the provided name
+ * @since unreleased
+ */
+export function getCustomInteger(
+  accessCredential: DatasetWithId,
+  field: URL,
+): number | undefined {
+  return deserializeField(
+    accessCredential,
+    field,
+    (i: unknown): i is number => typeof i === "number" && Number.isInteger(i),
+    "integer",
+  );
+}
+
+/**
+ * Reads the custom string array with the provided name in the consent section of the provided Access Credential.
+ * Throws on type mismatch, and returns an empty array if no values are found.
+ *
+ * @example
+ * ```
+ * const accessRequest = await issueAccessRequest({...}, {
+ *  ...,
+ *  customFields: new Set([
+ *     {
+ *       key: new URL("https://example.org/ns/customString"),
+ *       value: ["custom value", "another value"],
+ *     },
+ *   ]),
+ * });
+ * // s is ["custom value", "another value"]
+ * const s = getCustomStrings(accessRequest, new URL("https://example.org/ns/customString"));
+ * ```
+ *
+ * @param accessCredential The Access Credential (Access Grant or Access Request)
+ * @returns the custom string array field with the provided name
+ * @since unreleased
+ */
+export function getCustomStrings(vc: DatasetWithId, field: URL): string[] {
+  return deserializeFields(vc, field, (s) => typeof s === "string", "string");
```

Comment:
```suggestion
  return deserializeFields(
    accessCredential,
    field,
    (s) => typeof s === "string",
    "string",
  );  
```
Suggesting same layout but at a minimum, the param name needs to match